mod-source-record-storage-server/src/main/java/org/folio/services/handlers/actions/AbstractDeleteEventHandler.java

```diff
@@ -66,8 +68,10 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload p
   /* Handles DELETE action  */
   private void handlePayload(DataImportEventPayload payload, CompletableFuture<DataImportEventPayload> future) {
     var payloadRecord = Json.decodeValue(payload.getContext().get(getRecordKey()), Record.class);
+    var okapiHeaders = toOkapiHeaders(payload);
     LOG.info("handlePayload:: Handling 'delete' event for the record id = {}", payloadRecord.getId());
-    recordService.updateRecordsState(payloadRecord.getMatchedId(), RecordState.DELETED, RecordType.MARC_AUTHORITY, payload.getTenant())
+    recordService.deleteRecordById(payloadRecord.getMatchedId(), IdType.RECORD, okapiHeaders)
+      .recover(throwable -> throwable instanceof NotFoundException ? Future.succeededFuture() : Future.failedFuture(throwable))
```

Comment:
@RomanChernetskyi  Let's log here smth. At the debug level - that record was not found (with specific Id). And warn level - with an error message and cause.